Source/WTF/wtf/glib/RunLoopGLib.cpp

```diff
@@ -35,21 +36,68 @@
 #include <wtf/glib/ActivityObserver.h>
 #include <wtf/glib/RunLoopSourcePriority.h>
 
+#if HAVE(TIMERFD)
+#include <sys/timerfd.h>
+#include <time.h>
+#include <wtf/SystemTracing.h>
+#endif
+
 namespace WTF {
 
 typedef struct {
     GSource source;
     RunLoop* runLoop;
+#if HAVE(TIMERFD)
+    int timerFd;
+    struct itimerspec timerFdSpec;
+#endif
 } RunLoopSource;
 
 GSourceFuncs RunLoop::s_runLoopSourceFunctions = {
-    nullptr, // prepare
+    // prepare
+#if HAVE(TIMERFD)
+    [](GSource* source, int* timeout) -> gboolean {
+        auto& runLoopSource = *reinterpret_cast<RunLoopSource*>(source);
+
+        *timeout = -1;
+
+        if (runLoopSource.timerFd > -1) {
+            struct itimerspec timerFdSpec = { { 0, 0 }, { 0, 0 } };
```

Comment:
This could be a plain assignment to `= { }` to have the compiler zero-initialize the `struct`:


```suggestion
            struct itimerspec timerFdSpec = { };
```